Check for non-finite values in 'isPointInPathInternal' and 'isPointInStrokeInternal' functions

Ahmad-S792 opened the pull request:
#### 90fc0225ff3420961c296bf972f111c598e23d03
<pre>
Check for non-finite values in &apos;isPointInPathInternal&apos; and &apos;isPointInStrokeInternal&apos; functions

<a href="https://bugs.webkit.org/show_bug.cgi?id=250382">https://bugs.webkit.org/show_bug.cgi?id=250382</a>
rdar://problem/104331898

Reviewed by Said Abou-Hallawa.

This patch introduces check for non-finite values in &apos;isPointInPathInternal&apos; and &apos;isPointInStrokeInternal&apos; functions
by returning false. This is also extended to &apos;transformedPoint&apos; to have finite value by asserting above change.

* Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp:
(CanvasRenderingContext2DBase::isPointInPathInternal):
(CanvasRenderingContext2DBase::isPointInStrokeInternal):

Canonical link: <a href="https://commits.webkit.org/260946@main">https://commits.webkit.org/260946@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/84cbebe6a88c3d11e90dd3303fc3cc3e679fec86

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109999 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19099 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42681 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1442 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119045 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113952 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20565 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10292 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102257 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115745 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/15310 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98525 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43526 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97272 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30182 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85350 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11802 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31522 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12424 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8479 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17792 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51133 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7591 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14219 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp:
```diff
@@ -1155,6 +1155,9 @@ bool CanvasRenderingContext2DBase::isPointInStroke(Path2D& path, double x, doubl
 
 bool CanvasRenderingContext2DBase::isPointInPathInternal(const Path& path, double x, double y, CanvasFillRule windingRule)
 {
+    if (!std::isfinite(x) || !std::isfinite(y))
+        return false;
```
Please replace the check for the `transformedPoint` below:
```
  if (!std::isfinite(transformedPoint.x()) || !std::isfinite(transformedPoint.y()))
        return false;
```
with the assertion:
```
ASSERT(std::isfinite(transformedPoint.x()) && std::isfinite(transformedPoint.y());
```
If `x` and `y` are finite numbers, `transformedPoint.x()` and `transformedPoint.y()` have to be finite as well.

Ahmad-S792 replied:
Done.

shallawa commented on Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp:
```diff
@@ -1170,6 +1173,9 @@ bool CanvasRenderingContext2DBase::isPointInPathInternal(const Path& path, doubl
 
 bool CanvasRenderingContext2DBase::isPointInStrokeInternal(const Path& path, double x, double y)
 {
+    if (!std::isfinite(x) || !std::isfinite(y))
+        return false;
```
Ditto.

Ahmad-S792 replied:
Done.

Ahmad-S792 committed:
Check for non-finite values in 'isPointInPathInternal' and 'isPointInStrokeInternal' functions

https://bugs.webkit.org/show_bug.cgi?id=250382
rdar://problem/104331898

Reviewed by Said Abou-Hallawa.

This patch introduces check for non-finite values in 'isPointInPathInternal' and 'isPointInStrokeInternal' functions
by returning false. This is also extended to 'transformedPoint' to have finite value by asserting above change.

* Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp:
(CanvasRenderingContext2DBase::isPointInPathInternal):
(CanvasRenderingContext2DBase::isPointInStrokeInternal):

Canonical link: https://commits.webkit.org/260946@main
